browser/extensions/api/brave_rewards_api.cc

```diff
@@ -926,30 +927,30 @@ BraveRewardsGetAllNotificationsFunction::Run() {
   RewardsService* rewards_service =
     RewardsServiceFactory::GetForProfile(profile);
 
-  base::ListValue list;
+  base::Value::List list;
 
   if (!rewards_service) {
-    return RespondNow(OneArgument(std::move(list)));
+    return RespondNow(OneArgument(base::Value(std::move(list))));
   }
 
   auto notifications = rewards_service->GetAllNotifications();
 
   for (auto const& notification : notifications) {
-    auto item = std::make_unique<base::DictionaryValue>();
-    item->SetString("id", notification.second.id_);
-    item->SetInteger("type", notification.second.type_);
-    item->SetInteger("timestamp", notification.second.timestamp_);
+    base::Value::Dict item;
+    item.Set("id", notification.second.id_);
+    item.Set("type", notification.second.type_);
+    item.Set("timestamp", static_cast<int>(notification.second.timestamp_));
```

Comment:
@zenparsing: Is it ok that we are casting from int64_t to int here?

Reply:
Possibly we should use `base::Int64ToValue` there, and avoid the conversion altogether.

Reply:
Right, this is what I did in https://github.com/brave/brave-core/pull/13498/commits/5c71e38347a7a45f7e50d3b868716aac90f7a846#diff-710cdcbdcd014b58902a237fb17ec8e0e5594ce22a58ca182df1e35d68f637bdR232 and its counterpart in https://github.com/brave/brave-core/pull/13498/commits/5c71e38347a7a45f7e50d3b868716aac90f7a846#diff-710cdcbdcd014b58902a237fb17ec8e0e5594ce22a58ca182df1e35d68f637bdR190-R191, but in here these params are being sent to JS, which, from what I understand, doesn't have an equivalent of int64_t. We'd also have to adjust the json definition files to indicate that his value would become a string instead of an integer.

Reply:
Apologies for not seeing this before. I see that we had a lot of implicit `uint64_t` to `int` conversions happening and you've made them into explicit casts to `int`.

I think the `uint64_t` conversions to `int` are undesirable. Since JS doesn't know about ints (only doubles), I think it might make the most sense to set all these values as doubles instead of int.

Reply:
@zenparsing I think that's a great idea. It would be better to solve these outside the scope of the Chromium bump.

Reply:
Created https://github.com/brave/brave-browser/issues/24015